x-pack/plugins/enterprise_search/public/applications/__mocks__/shallow_with_i18n.mock.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import { I18nProvider } from '@kbn/i18n/react';
+import { IntlProvider } from 'react-intl';
+
+const intlProvider = new IntlProvider({ locale: 'en', messages: {} }, {});
+const { intl } = intlProvider.getChildContext();
+
+/**
+ * This helper shallow wraps a component with react-intl's <I18nProvider> which
+ * fixes "Could not find required `intl` object" console errors when running tests
+ *
+ * Example usage (should be the same as shallow()):
+ *
+ * const wrapper = shallowWithIntl(<Component />);
+ */
+export const shallowWithIntl = children => {
+  return shallow(<I18nProvider>{children}</I18nProvider>, {
+    context: { intl },
+    childContextTypes: { intl },
+  })
+    .childAt(0)
+    .dive()
+    .shallow();
```

Comment:
This returns the underlying `<Child />` component instead of `(<I18nProvider><Child /></I18nProvider>`, which makes `.find()`ing children less of a pain.

I had to do this a bunch on older projects for HOC wrappers, I can 10000% say I don't miss it 😄